emmanuelmacharia/Questioner-Database-Integration
# Questioner- database Integration and Testing

## What does this Pull Request Do? ##
rewrote the [Questioner](https://github.com/emmanuelmacharia/Questioner) application; Not with PostgreSQL and psycopg2 integration built in. Connects to three main databases depending on the environments within which the application is running- these environments are:
- development - this is active during the entire development cycle of the application
-  testing - when running tests, this database is used
-  production - when the application is ready, this is the database the app will connect to
This PR also creates endpoints and their respective models and contains the tests for these endpoints.

## Description of the task to be completed ##
Integrate databases into the questioner application using `psycopg2` and have tests working for the application

## How should this be manually tested? ##

1. `clone` this repository onto your local computer
2. Create a virtual environment and use `pip install requirements` within the virtual env
3. run these commands: `export FLASK_APP="run.py"` , `export APP_SETTINGS="development"` 
4. You will need to create a development and test database; you can [refer here](https://www.digitalocean.com/community/tutorials/how-to-install-and-use-postgresql-on-ubuntu-18-04#creating-a-new-role) if on Linux or [here](http://www.techken.in/linux/install-postgresql-10-windows-10-linux/) if you are using Windows.
5. run these commands to connect to the databases you created in step 4: 
- `export Dev_URL="dbname=dbname user=username password=password"`
- `export Test_URL="dbname=dbname user=username password=password"`
- `export SECRET="YourSecretKeyGoesHere"`
6. Run the tests using `pytest` and/or `coverage`. If you'd like to see more detailed tests on the code, run `coverage html` and navigate to the htmlcov folder created in your main directory then run the `index.html` file. ALL TESTS SHOULD PASS

 

 

 

##
